### ISSUE TITLE
[BUG] Sluggish scrolling on galleries with high res pics

### ISSUE THREAD

**Issue by putrasattvika** [laurencedawson/sync-for-reddit#1111]
**Describe the bug**
> _Based on this thread on r/redditsync: https://www.reddit.com/r/redditsync/comments/tkl8xy/bug_v22_sluggish_scrolling_on_galleries_with_high/_

There are two identifiable image-related performance affecting Sync v22:
- Sluggish image swiping when viewing Reddit/imgur galleries when the images are sufficiently high-resolution. Example: https://www.reddit.com/r/cats/comments/tk2hgq/_/ (each image is ~3000x4000 pixels).
- Sluggish post scrolling with multiple embedded high-resolution images. Example: https://www.reddit.com/r/apexlegends/comments/tjhhfh/a_twitch_streamer_is_boosting_his_account_by/ (each image is ~1080p).

Whether or not this issue affects older Sync versions is unknown.

This might have something to do with Samsung's Android 12 skin (One UI 4.x), as both my and OP's devices are Samsung running Android 12. As far as I remember, I have never encountered this issue while my device is still running Android 11.

Also, I don't think this has anything to do with the device's performance as OP's device (Galaxy 20FE) is quite powerful, and even my 4 years old mid-range phone (running Android 9) can handle the gallery/post smoothly.

**To Reproduce**
1. Go to either [this post for sluggish gallery swiping](https://www.reddit.com/r/cats/comments/tk2hgq/_/) or [this post for sluggish post scrolling](https://www.reddit.com/r/apexlegends/comments/tjhhfh/a_twitch_streamer_is_boosting_his_account_by/)
2. Swipe the gallery or scroll down the post

**Expected behavior**
Scrolling should be smooth.

**Screenshots**
Video proof, taken from the original Reddit bug report: https://imgur.com/a/k4KOD5P

**Smartphone (please complete the following information):**
Reddit thread OP's device:
 - Device: Samsung Galaxy S20 FE (SM-G780F)
 - OS: Android 12
 - Browser: Unknown
 - Version: v22.3.21 (dev)

My affected device:
 - Device: Samsung Galaxy Tab S6 Lite (SM-P615)
 - OS: Android 12 (One UI 4.1)
 - Browser: [Fennec F-Droid](https://f-droid.org/en/packages/org.mozilla.fennec_fdroid/)
 - Version: v22.4.20 (beta)

**Comment by laurencedawson** [comment]
Hi,
If you go to Settings > Images > Gallery DPI and adjust this to 200 this will fix your issue.

Cheers

**Comment by putrasattvika** [comment]
> Hi, If you go to Settings > Images > Gallery DPI and adjust this to 200 this will fix your issue.

Unfortunately this does not solve the issue in my case, even after clearing image cache & application cache. Even changing the value to the highest value possible (400 dpi) does _not_ make the swiping even more sluggish; the sluggishness I perceive does not change between DPI values.

I believe that it is not the hardware's fault as swiping between high-res images using an [external gallery app](https://github.com/SimpleMobileTools/Simple-Gallery) works just fine, but I understand that this issue might be very hard to troubleshoot considering the lack of affected users.

I don't have non-Android 12 Samsung devices nor non-Samsung Android 12 devices so I can't confirm whether this affects all Android 12 devices or just Samsung's.

**Comment by putrasattvika** [comment]
Tested older versions of Sync free (from APKMirror) and this issue affects v20 _but not_ v19. I also noticed that scrolling through the feed (cards view, 2 columns) is also relatively stutter-free on v19 (compared to v22 which will noticeably stutter).

Considering that this seems like a performance regression and v22 will replace v19 on the stable channel soon, I think we should revisit this issue before the final release of v22.
